### ISSUE TITLE
Run result are different between version 1.0.2 and 2.0

### ISSUE THREAD

**Issue by kaiwang0112006** [MSeal/agglom_cluster#12]
I have tyied both [version 1.0.2](https://pypi.python.org/pypi/AgglomCluster/1.0.2) and [version 2.0.1](https://pypi.python.org/pypi/AgglomCluster/2.0.1) to analyze my graph. The results are different.

For version 1.0.2, there's only a .py file and I just import it into my code:

    newman = NewmanGreedy(graph)
    cluster_set = newman.get_clusters()

it will return a set of subgraphs that are good enough for me.

While for version 2.0.1, I have to pip install it, and the usage are different:

    from hac import GreedyAgglomerativeClusterer
    clusterer = GreedyAgglomerativeClusterer()
    karate_dendrogram = clusterer.cluster(graph)
    print(karate_dendrogram.clusters())

The result set is empty...

**Comment by MSeal** [comment]
The `pip install agglomcluster==2.0.1` is a standard method of installing dependencies in python. You can also download source code and run `python setup.py install`. Having a py file you copy into a project is mostly a coincidence rather than the norm. The interface change was parried with changing the major version from 1 to 2. See http://semver.org/ for expectations with version changes for most code repositories.

What's the graph object have in it? Was that the karate graph as it implies or a custom graph? I've used that version for a while now and my results did not change with any project I used. Here's the karate example on python 2.7 with version 2.0.1 on a windows machine:

```
import networkx as nx
graph = nx.karate_club_graph()

from hac import GreedyAgglomerativeClusterer
clusterer = GreedyAgglomerativeClusterer()
karate_dendrogram = clusterer.cluster(graph)
print(karate_dendrogram.clusters())
// => [set([32, 33, 8, 14, 15, 18, 20, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31]), set([1, 2, 3, 7, 9, 12, 13, 17, 21]), set([0, 4, 5, 6, 10, 11, 16, 19])]
```

**Comment by kaiwang0112006** [comment]
It's odd to me too that results are different. My graph is undirected and contain only words and connections mean two words have similar meaning. I've pickled my graph [here](https://github.com/kaiwang0112006/mytranfor/blob/master/graph.pkl). It's generated by networkx.

**Comment by MSeal** [comment]
That is odd, I won't be able to look into it for a few days. I'll try loading that graph you posted and see if there's a bug introduced or something else wrong.

**Comment by MSeal** [comment]
I was able to reproduce the issue and debug it a little. It looks like the right number of clusters is selected but it ends up not populating the return object for some reason. I'll do another debug pass this week and should have a fix by next weekend.

**Comment by kaiwang0112006** [comment]
ok, Great~

**Comment by MSeal** [comment]
Ok, I believe the issue you discovered is fixed in v2.0.4 now.

Python27 code:
```
import pickle
from hac import GreedyAgglomerativeClusterer

with open("graph.pkl", 'rb') as pfile:
    graph = pickle.load(pfile)

clusterer = GreedyAgglomerativeClusterer()
dendrogram = clusterer.cluster(graph)
clust = dendrogram.clusters()
print(len(clust))
# 4796
```
